Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+            QueueDel(component.Observer.Value);
+        }
+
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+            blobTileComponent.Core = null;
+
+            blobTileComponent.Color = Color.White;
+            Dirty(blobTileComponent);
+        }
+    }
+
+    private void ChangeBlobEntChem(EntityUid uid, BlobChemType oldChem, BlobChemType newChem)
+    {
+        var explosionResistance = EnsureComp<ExplosionResistanceComponent>(uid);
+        if (oldChem == BlobChemType.ExplosiveLattice)
+        {
+            _explosionSystem.SetExplosionResistance(uid, 0.3f, explosionResistance);
+        }
+        switch (newChem)
+        {
+            case BlobChemType.ExplosiveLattice:
+                _damageable.SetDamageModifierSetId(uid, "ExplosiveLatticeBlob");
+                _explosionSystem.SetExplosionResistance(uid, 0f, explosionResistance);
+                break;
+            case BlobChemType.ElectromagneticWeb:
+                _damageable.SetDamageModifierSetId(uid, "ElectromagneticWebBlob");
+                break;
+            case BlobChemType.ReactiveSpines:
+                _damageable.SetDamageModifierSetId(uid, "ReactiveSpinesBlob");
+                break;
+            default:
+                _damageable.SetDamageModifierSetId(uid, "BaseBlob");
+                break;
+        }
+    }
+
+    public bool TransformBlobTile(EntityUid? oldTileUid, EntityUid coreTileUid, string newBlobTileProto,
+        EntityCoordinates coordinates, BlobCoreComponent? blobCore = null, bool returnCost = true,
+        FixedPoint2? transformCost = null)
+    {
+        if (!Resolve(coreTileUid, ref blobCore))
+            return false;
+        if (oldTileUid != null)
+        {
+            QueueDel(oldTileUid.Value);
+            blobCore.BlobTiles.Remove(oldTileUid.Value);
+        }
+        var tileBlob = EntityManager.SpawnEntity(newBlobTileProto, coordinates);
+
+        if (TryComp<BlobTileComponent>(tileBlob, out var blobTileComponent))
+        {
+            blobTileComponent.ReturnCost = returnCost;
+            blobTileComponent.Core = coreTileUid;
+            blobTileComponent.Color = blobCore.ChemСolors[blobCore.CurrentChem];
+            Dirty(blobTileComponent);
+
+            var explosionResistance = EnsureComp<ExplosionResistanceComponent>(tileBlob);
+
+            if (blobCore.CurrentChem == BlobChemType.ExplosiveLattice)
+            {
+                _explosionSystem.SetExplosionResistance(tileBlob, 0f, explosionResistance);
+            }
+        }
+        if (blobCore.Observer != null && transformCost != null)
+        {
+            _popup.PopupEntity(Loc.GetString("blob-spent-resource", ("point", transformCost)),
+                tileBlob,
+                blobCore.Observer.Value,
+                PopupType.LargeCaution);
+        }
+        blobCore.BlobTiles.Add(tileBlob);
```

Comment:
Add the tile to the list sooner after spawning than this, so if any code here throws the tile is still added.